numpy/_core/include/numpy/ndarraytypes.h

```diff
@@ -1239,7 +1239,18 @@ typedef struct {
         npy_intp             index;                   /* current index */
         int                  nd;                      /* number of dims */
         npy_intp             dimensions[NPY_MAXDIMS_LEGACY_ITERS]; /* dimensions */
-        PyArrayIterObject    *iters[NPY_MAXARGS];     /* iterators */
+        /*
+         * Space for the indivdual iterators, do not specify size publically
+         * to allow changing it more easily.
+         * One reason is that Cython uses this for checks and only allows
+         * growing structs (as of Cython 3.0.6).  It also allows NPY_MAXARGS
+         * to be runtime dependent.
+         */
+#if defined(NPY_INTERNAL_BUILD) && NPY_INTERNAL_BUILD
+        PyArrayIterObject    *iters[64];  /* 64 is NPY_MAXARGS */
```

Comment:
The compat header is included later, so it could use these types in principle...